dali/operators/reader/loader/video/frames_decoder_gpu.cc

```diff
@@ -260,9 +260,16 @@ BufferedFrame& FramesDecoderGpu::FindEmptySlot() {
   DALI_FAIL("Could not find empty slot in the frame buffer");
 }
 
-
 void FramesDecoderGpu::Reset() {
   SendLastPacket(true);
   FramesDecoder::Reset();
 }
+
+FramesDecoderGpu::~FramesDecoderGpu() {
+  cuvidDestroyVideoParser(nvdecode_state_->parser);
+  cuvidDestroyDecoder(nvdecode_state_->decoder);
+
```

Comment:
BTW is there a reason to not using `cuvideodecoder.h` `cuvideoparser.h` that are in the code base? Just asking, not saying we should be using it.

Reply:
The old `VideoReader` and all of its utils generally rely on older version of NVCUVID. I plan to remove them all, when new reader is fully operational. 
General architecture idea here is for `FramesDecoder` and `FramesDecoderGpu` to hide the whole FFMPEG and Video Decode SDK from the rest of the code, but inside I found it easier to work with them directly rather than wrapping various parts of those libraries.